src/advanced_agent/learning/prompt_manager.py

```
@@ -0,0 +1,909 @@
+            # 変数の整合性チェック
+            extracted_vars = self.extract_variables()
+            if self.variables and set(self.variables) != set(extracted_vars):
+                validation_result["warnings"].append("Variable list doesn't match template variables")
+            
+            # テンプレートの構文チェック
+            try:
+                # サンプルデータでレンダリングテスト
+                sample_data = {var: f"sample_{var}" for var in extracted_vars}
+                self.render(**sample_data)
+                validation_result["score"] += 20
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)